### PR TITLE
Sound music

### PR DESCRIPTION
**Pull request by MeteoricDragon**
Implements Music volume control in DFU.  Sound volume control is implemented but not tied to anything because I'm not sure where the most appropriate place(s) to put the global sound multiplier is.  (do we want to scale *all* sound effects?

### CODE REVIEW

**Comment by MeteoricDragon** [comment]
Ok, that's weird.  One of the commits is me merging your master file into my fork's branch SoundMusic.  Is that harmless?

**Comment by Interkarma** [comment]
> Ok, that's weird. One of the commits is me merging your master file into my fork's branch SoundMusic. Is that harmless?

That should be fine. I'm only receiving the changed files in the PR. :)

File: Assets/Scripts/Game/UserInterfaceWindows/DaggerfallPauseOptionsWindow.cs [hunk 1]
```diff
@@ -77,6 +81,18 @@ protected override void Setup()
             //loadButton.BackgroundColor = DaggerfallUI.DaggerfallUnityNotImplementedColor;
             loadButton.OnMouseClick += LoadButton_OnMouseClick;
 
+            // Sound Bar
+            Button soundPanel = DaggerfallUI.AddButton(new Rect(6.15f, 23.20f, barMaxLength, 5.5f), optionsPanel);
```
Comment (Interkarma) [inline Assets/Scripts/Game/UserInterfaceWindows/DaggerfallPauseOptionsWindow.cs]: You shouldn't need to use part-pixel coordinates here, but I know why you had to. This is due to a scaling bug when Daggerfall-native canvases are scaled and centred in parent screen-native canvas. I'll try to fix this after merging PR.

**Comment by Interkarma** [comment]
Thanks for making a start on this, looks good!

> I'm not sure where the most appropriate place(s) to put the global sound multiplier is. (do we want to scale all sound effects?

I think that would be the case. Best place would be in DaggerfallAudioSource.PlayOneShot() and multiply volumeScale by global volume level. For example at line 207:

    audioSource.PlayOneShot(clip, volumeScale * DaggerfallUnity.Settings.SoundVolume);

There might be some edge cases where sound isn't fed through DaggerfallAudioSource.PlayOneShot() that I can't recall. I think around 99% of Daggerfall's sounds should be taken care of with the above change.

**Comment by MeteoricDragon** [comment]
I got a whole bunch of them, but Torches on walls I can't seem to find.

**Comment by Interkarma** [comment]
I knew I was forgetting something. Try adding the following right under line 264 of DaggerfallAudioSource.cs

    audioSource.volume = DaggerfallUnity.Settings.SoundVolume;

That should control the volume for looping effects and proximity effects like torches. I don't think this will conflict with the PlayOneShot() method as the oneshot methods pull their own version of the clip.

Let me know how you go!

**Comment by Interkarma** [comment]
There's a bunch of headbobber stuff in there as well - things I merged earlier. Don't forget to update your branch to current master.

File: Assets/Scripts/Game/DaggerfallUI.cs [hunk 2]
```diff
@@ -622,7 +622,7 @@ public AudioClip GetAudioClip(SoundClips clip)
         public void PlayOneShot(AudioClip clip)
         {
             if (audioSource)
-                audioSource.PlayOneShot(clip);
+                audioSource.PlayOneShot(clip, DaggerfallUnity.Settings.SoundVolume);
```
Comment (Interkarma) [inline Assets/Scripts/Game/DaggerfallUI.cs]: There's no need to multiply the volume in all these places - everything goes through DaggerfallAudioSource anyway.

Please try my earlier suggestion to set volume in DaggerfallAudioSource right after spatialBlend. It should only be 2x lines of code and should cover everything.

**Comment by MeteoricDragon** [comment]
Fixed it.  I don't see anything about the headbobber stuff in the soundmusic branch.  I've merged your master to my master, and soundMusic branch.  Also, your suggestion was already implemented in the DaggerfallAudioSource.  I just didn't realize that it would make all the other multiplications of the Sound setting in other files unneccessary.

**Comment by Interkarma** [comment]
That looks much better, cheers mate. :)

Hopefully that covers everything. I'll test and merge as soon as I can.

**Comment by MeteoricDragon** [comment]
That's really weird that it would add those old commits.  @TheLacus I typed the commands exactly as you suggested, except I removed "HEAD" from `git push origin:HEAD SoundMusic --force` because git was giving me an error: 

> $ git push origin:HEAD SoundMusic --force
ssh: Could not resolve hostname origin: Name or service not known
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.

**Comment by MeteoricDragon** [comment]
I've attempted several ways to clean the commits using the Git advice here and each attempt causes the PR  include new commits to undo commits that were caused in my branch from merging upstream master code from Interkarma's repository several days ago and on more than one occasion. I know one way I can submit a clean PR is to copy and paste the code from this PR and make a new branch with that code and publish and submit it for new PR.  Will the code commits I have on this PR work?  Whenever you are ready, my branch is ready unless you wish for me to redo it.

**Comment by TheLacus** [comment]
> I know one way I can submit a clean PR is to copy and paste the code from this PR and make a new branch with that code and publish and submit it for new PR. Will the code commits I have on this PR work?

You have to pull the new commits to have the changes on your local branch.
```
git pull --rebase upstream master 
```
or
```
git fetch upstream
git rebase upstream master
```

upstream is Interkarma's repository, master is the name of the branch.

**Comment by Allofich** [comment]
> except I removed "HEAD" from git push origin:HEAD SoundMusic --force because git was giving me an error:

Sorry the colon was in the wrong spot. It should have been `git push origin HEAD:SoundMusic --force`. But I guess `HEAD` was unnecessary as you found out.

**Comment by Interkarma** [comment]
This works really well! One thing I noticed is that player footsteps don't seem to scale volume even though using PlayOneShot() and the volume scale is applied. I didn't have time this morning to dig into this. Will look into this later if you don't beat me to it.

I also noticed it's a bit hard to get the slider back to max position, it always seems to stop just one notch short. One easy fix would be to snap value to 1.0 once over a certain threshold.

Good job :)

**Pull request merged**